lib/view_component/slotable_v2.rb

```diff
@@ -250,6 +256,10 @@ def set_slot(slot_name, *args, **kwargs, &block)
 
       if slot_definition[:collection]
         @__vc_set_slots[slot_name] ||= []
+
+        # set the index to the next index in the array (aka, the current length)
+        slot.slot_index = @__vc_set_slots[slot_name].length
```

Comment:
What happens if we sort the collection before rendering?

Reply:
:/ Yeah, that wouldn't work with the current implementation. I'm not sure if that'd be a common use-case or not though. It's certainly not for me.

Reply:
From my experience, slot introspection is not out of the ordinary. With that in mind, do you think we should still pursue this feature?

Reply:
I definitely would still find value in this feature. Since creating this PR, I've run into a couple other spots where it would be useful. I just haven't found the time to make the improvements to it yet 😓

Reply:
I would find it useful as well. I have a few components where it would help.